src/reconcile/rec_write.c

```diff
@@ -2369,6 +2369,33 @@ __rec_split_dump_keys(WT_SESSION_IMPL *session, WT_RECONCILE *r)
     return (0);
 }
 
+/*
+ * __rec_page_modify_ta_safe_free --
+ *     Any thread that is reviewing the page modify time aggregate in a WT_REF, must also be holding
+ *     a split generation to ensure that the page index they are using remains valid. Use that same
+ *     split generation to ensure that the page modify time aggregate inside the WT_REF remains
+ *     valid while it is being reviewed.
+ */
+static void
+__rec_page_modify_ta_safe_free(WT_SESSION_IMPL *session, WT_TIME_AGGREGATE **ta)
```

Comment:
Why we need to free it like this way? Can this be freed when the entire modify structure is gone?

Reply:
If we do not free the timestamp information, all the search operations are unnecessarily check for the visibility even if there valid entries on the page. This is to speed up the search.

Reply:
In that case, the page should be dirty and we never check this?

Reply:
This function is called in the flow of reconciliation. There is no need to check for it.